feat(curriculum): add HTML video player project to FSD cert

mtrivera opened the pull request:
Checklist:

<!-- Please follow this checklist and put an x in each of the boxes, like this: [x]. It will ensure that our team takes your pull request seriously. -->

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitHub Codespaces.

<!-- Feel free to add any additional description of changes below this line -->

Closes freeCodeCamp/CurriculumExpansion#841

mtrivera committed:
build: Scaffold workshop-html-video-player project

mtrivera committed:
feat: add step 1 instructions and step 2 seed code

mtrivera committed:
feat: add step 2 instructions and step 3 seed code

mtrivera committed:
fix: Remove solution in step 2

mtrivera committed:
feat: Add step 3 instructions and step 4 seed code

mtrivera committed:
feat: Add step 4 instructions and step 5 seed code

mtrivera committed:
feat: Add step 5 instructions and step 6 seed code

mtrivera committed:
feat: Add step 6 instructions and step 7 seed code

mtrivera committed:
feat: Add step 7 instructions and step 8 seed code

majestic-owl448 reviewed (commented):
please make sure the title does match the guidelines

ilenia-magoni commented on client/i18n/locales/english/intro.json:
```diff
@@ -5004,6 +5004,10 @@
         "title": "Certified Full Stack Developer Exam",
         "intro": ["Pass this exam to become a Certified Full Stack Developer."]
+      },
+      "workshop-html-video-player": {
```
please move this upward so that it is below `lecture-working-with-audio-and-video-elements`

mtrivera committed:
feat: Add step 8 instructions and step 9 seed code

mtrivera committed:
feat: Add step 9 instructions and step 10 seed code

mtrivera committed:
feat: Add step 10 instructions and step 11 seed code

mtrivera committed:
feat: Add step 11 instructions and step 12 code

mtrivera committed:
feat: Add step 12 instructions and step 13 seed code

mtrivera committed:
feat: Add step 13 instructions and step 14 seed code

mtrivera committed:
fix: Add missing seed data for step 14

mtrivera committed:
feat: Add step 14 instructions and step 15 seed code

mtrivera committed:
feat: Add last step directions and final solution code

ilenia-magoni commented on curriculum/challenges/english/blocks/workshop-html-video-player/68f1f371fefd448b4f70b898.md:
```diff
@@ -0,0 +1,46 @@
+
+# --description--
+
+In the HTML Video and Audio Element Theory reading,
```
```suggestion
In a previous lesson,
```

we don't want to have to update multiple places if we need to change a title

ilenia-magoni commented on curriculum/challenges/english/blocks/workshop-html-video-player/68f27139bddf9f6bf2821acf.md:
```diff
@@ -0,0 +1,66 @@
+
+# --description--
+
+Let's add one final `source` element. This one will use the `video/quicktime` MIME type.
```
you use "let's" often, remember that it stans for "Let us", and you should not use first person in the text

jdwilkin4 committed:
Apply suggestions from code review

Co-authored-by: Ilenia <26656284+ilenia-magoni@users.noreply.github.com>

mtrivera committed:
chore: Move workshop-html-video-player

Move workshop-html-video-player above workshop-music-player

mtrivera committed:
docs: Add body/html closing tags to step 1 seed code

mtrivera committed:
style: Clean wording and typo in step 1 description

mtrivera committed:
test: Add step 1 hint and test

mtrivera committed:
build: Remove deprecated step 15

mtrivera committed:
build: Remove deprecated step 14

mtrivera committed:
test: Add step 2 hint and test

mtrivera committed:
chore: Add body/closing tags to step 2 seed code

mtrivera committed:
docs: Cleanup description wording in step 3

mtrivera committed:
docs: Cleanup description wording in step 3

mtrivera committed:
fix: Make video element editable in step 3 seed code

mtrivera committed:
test: Add step 3 hint and test

mtrivera committed:
test: Add step 4 hint and test

mtrivera committed:
fix: Add body/html closing tags in step 4 seed code

mtrivera committed:
fix: Cleanup step 5 seed code

Wrap the video element in the editable region block. Add body/html closing tags.

mtrivera committed:
test: Add step 5 hint and test

mtrivera committed:
fix: Wrap video in editable block step 4

mtrivera committed:
fix: Add closing tags and adjust edit block for step 6

mtrivera committed:
test: Add step 6 hint and test

mtrivera committed:
fix: Adjust edit block and add closing tags step 7

mtrivera committed:
test: Add step 7 hint and test

mtrivera committed:
fix: Add closing tags to step 8 seed code

mtrivera committed:
test: Add step 8 hint and test

mtrivera committed:
fix: Adjust edit block and add closing tags step 9

mtrivera committed:
test: Add step 9 hint and test

mtrivera committed:
fix: Adjust edit block and add closing tags step 10

mtrivera committed:
docs: Remove external link and insert content instead

mtrivera committed:
test: Add step 10 hint and test

mtrivera committed:
docs: Specify video/mp4 type in description step 10

mtrivera committed:
docs: Specify video/mp4 type in description step 10

mtrivera committed:
fix: Add body/html closing tags to step 11 seed code

mtrivera committed:
test: Add step 11 hint and test

mtrivera committed:
docs: Change voice of description step 11

mtrivera committed:
fix: Add body/html closing tags to seed code step 12

mtrivera committed:
test: Add step 12 hint and test

mtrivera committed:
docs: Change the voice of the description block step 12

mtrivera committed:
docs: Add solutions and fix seed code final step

mtrivera committed:
test: Add final step hint and test

mtrivera committed:
docs: Change voice and add completion message to final step

mtrivera committed:
Merge branch 'html-video-workshop-841' of github.com:mtrivera/freeCodeCamp into html-video-workshop-841

mtrivera committed:
fix: Remove extra head clsing tag in step 1 seed code

majestic-owl448 reviewed (commented):
please make sure the title follows the guidelines and make sure to link the issue in the first post

ilenia-magoni commented on curriculum/challenges/english/blocks/workshop-html-video-player/68f268880219c2485bd71d2a.md:
```diff
@@ -0,0 +1,58 @@
+
+# --description--
+
+Now that we have our `source` element, we can add `src` attribute.
```
don't use 'we', only 'you'

mtrivera committed:
test: Change to boolean assertion step 5

mtrivera committed:
test: Change to boolean assertion step 6

mtrivera committed:
docs: Change voice in description step 8

mtrivera committed:
docs: Change the description voice of step 9

mtrivera committed:
docs: Change description voice step 1

mtrivera committed:
style: Add period to hint text step 1

mtrivera committed:
docs: Improve description grammar of step 7

ilenia-magoni commented on curriculum/challenges/english/blocks/workshop-html-video-player/68f18724d71db1dda8d6fec7.md:
```diff
@@ -0,0 +1,42 @@
+title: Step 1
+challengeType: 0
+dashedName: step-1
+demoType: onClick
```
```suggestion
demoType: onLoad
```

jdwilkin4 committed:
Update curriculum/challenges/english/blocks/workshop-html-video-player/68f18724d71db1dda8d6fec7.md

Co-authored-by: Ilenia <26656284+ilenia-magoni@users.noreply.github.com>

mtrivera committed:
fix: Move workshop block in intro.json

mtrivera committed:
chore: Change 'blockType' to 'blockLabel'

ilenia-magoni commented on curriculum/challenges/english/blocks/workshop-html-video-player/68f18724d71db1dda8d6fec7.md:
````diff
@@ -0,0 +1,42 @@
+
+```js
+const h1 = document.querySelector('h1');
+assert.strictEqual(h1?.textContent, 'Working with the HTML Video Element');
````
```suggestion
assert.strictEqual(h1?.textContent.trim(), 'Working with the HTML Video Element');
```

ilenia-magoni commented on curriculum/challenges/english/blocks/workshop-html-video-player/68f1f8c248306da204655117.md:
```diff
@@ -0,0 +1,51 @@
+Omitting the `muted` attribute will play
+audio on initial playback.
+
+Now, add the `muted` attribute to the `video` element.
```
```suggestion
The `muted` attribute will silence audio
on initial playback. If you have `controls`
enabled, the user will be able to unmute audio.
Omitting the `muted` attribute will play
audio on initial playback.

The `muted` attribute is a boolean attribute, it does not need a value.

Now, add the `muted` attribute to the `video` element.
```

ilenia-magoni commented on curriculum/challenges/english/blocks/workshop-html-video-player/68f1f6f97af9749a42260a43.md:
```diff
@@ -0,0 +1,49 @@
+volume control, etc. Omitting the `controls` will not
+offer user controls.
+
+Now, add the `controls` attribute to the `video` element.
```
```suggestion
The `controls` attribute provides playback controls.
Think of a physical remote control: playback, rewind,
volume control, etc. Omitting the `controls` will not
offer user controls.

The `controls` attribute is a boolean attribute, it does not need a value.

Now, add the `controls` attribute to the `video` element.
```

ilenia-magoni commented on curriculum/challenges/english/blocks/workshop-html-video-player/68f268880219c2485bd71d2a.md:
```diff
@@ -0,0 +1,58 @@
+Add the `src` attribute and the value
+`https://cdn.freecodecamp.org/curriculum/labs/what-is-the-map-method-and-how-does-it-work.mp4`.
+
+Notice, the file extension `mp4`. You will learn about it in the next lesson.
```
```suggestion
Now that you inserted the `source` element, you will add the `src` attribute.
It the specifies the URL of a media resource.

Add the `src` attribute with the value
`https://cdn.freecodecamp.org/curriculum/labs/what-is-the-map-method-and-how-does-it-work.mp4`.
```

ilenia-magoni commented on curriculum/challenges/english/blocks/workshop-html-video-player/68f26d52b13e0f5b6e2d2e09.md:
````diff
@@ -0,0 +1,65 @@
+assert.strictEqual(source?.getAttribute('type'), 'video/webm');
+```
+
+# --seed--
````
````suggestion
# --hints--

You should have a second `source` eleemnt.

```js
assert.exists(document.querySelector('video > source:nth-of-type(2)'));
```

Your second `source` element should have a `src` attribute with a value of `https://cdn.freecodecamp.org/curriculum/labs/mapmethod.webm`.

```js
const source = document.querySelector('source:nth-of-type(2)');
assert.strictEqual(source?.getAttribute('src'), 'https://cdn.freecodecamp.org/curriculum/labs/mapmethod.webm');
```

Your second `source` element should have a `type` attribute with a value of `video/webm`.

```js
const source = document.querySelector('source:nth-of-type(2)');
assert.strictEqual(source?.getAttribute('type'), 'video/webm');
```

# --seed--
````

ilenia-magoni commented on curriculum/challenges/english/blocks/workshop-html-video-player/68f26a5f79e97c5032c10642.md:
````diff
@@ -0,0 +1,61 @@
+```js
+const source = document.querySelector('source');
+assert.strictEqual(source?. getAttribute('type'), 'video/mp4');
+```
````
````suggestion
Your `source` element should have a `type` attribute with the value `video/mp4`.

```js
const source = document.querySelector('source');
assert.strictEqual(source?. getAttribute('type'), 'video/mp4');
```
````

mtrivera committed:
docs: step 1 description style and grammar

Apply step 1 description style and grammar suggestion from @ilenia-magoni

Co-authored-by: Ilenia <26656284+ilenia-magoni@users.noreply.github.com>

mtrivera committed:
test: Add trim method to h1 text content check for step 1

Add trim method to h1 text content check for step 1 suggestion from @ilenia-magoni

Co-authored-by: Ilenia <26656284+ilenia-magoni@users.noreply.github.com>

mtrivera committed:
style(curriculum): Apply suggestions from code review

Improve readability of workshop steps

Co-authored-by: Ilenia <26656284+ilenia-magoni@users.noreply.github.com>

mtrivera committed:
style: Make the word 'pixel' plural

mtrivera committed:
docs: Remove gif from description step 4

mtrivera committed:
style: Improve reminder text for step 8

mtrivera committed:
style: Fix element hint spelling of step 12

mtrivera committed:
style: Fix element hint spelling of step 13

mtrivera committed:
style: Remove deprecated 'the' from step 9 description

jdwilkin4 commented on curriculum/challenges/english/blocks/workshop-html-video-player/68f18724d71db1dda8d6fec7.md:
```diff
@@ -0,0 +1,44 @@
+
+You have been provided the HTML5 boilerplate code.
+
+First, create an `h1` element and give it the text `Working with the HTML Video Element`.
```
```suggestion
Create an `h1` element and give it the text `Working with the HTML Video Element`.
```

jdwilkin4 commented on curriculum/challenges/english/blocks/workshop-html-video-player/68f18724d71db1dda8d6fec7.md:
````diff
@@ -0,0 +1,44 @@
+```js
+const h1 = document.querySelector('h1');
+assert.strictEqual(h1?.textContent.trim(), 'Working with the HTML Video Element');
+```
````
split this into two tests

one to check if they have an h1

another to check for the h1 text

jdwilkin4 commented on curriculum/challenges/english/blocks/workshop-html-video-player/68f1f371fefd448b4f70b898.md:
```diff
@@ -0,0 +1,47 @@
+  <video
+
+  >
+--fcc-editable-region--
```
we don't need to have the opening tag open like this. they will know where to place the attributes as long as the editable regions are in the correct place 

```suggestion
--fcc-editable-region--
  <video>
--fcc-editable-region--
```

jdwilkin4 commented on curriculum/challenges/english/blocks/workshop-html-video-player/68f1f57723debf9383c75947.md:
```diff
@@ -0,0 +1,51 @@
+
+The `loop` attribute is a boolean attribute, it does not need a value.
+
+Now, add the `loop` attribute to the `video` element.
```
```suggestion
Add the `loop` attribute to the `video` element.
```

jdwilkin4 commented on curriculum/challenges/english/blocks/workshop-html-video-player/68f1f6f97af9749a42260a43.md:
```diff
@@ -0,0 +1,51 @@
+The `controls` attribute provides playback controls.
+Think of a physical remote control: playback, rewind,
+volume control, etc. Omitting the `controls` will not
+offer user controls.
```
```suggestion
The `controls` attribute provides playback controls including playback, rewind, and volume control for the `video` element. 
```

jdwilkin4 commented on curriculum/challenges/english/blocks/workshop-html-video-player/68f1f8c248306da204655117.md:
```diff
@@ -0,0 +1,53 @@
+
+The `muted` attribute is a boolean attribute, it does not need a value.
+
+Now, add the `muted` attribute to the `video` element.
```
```suggestion
Add the `muted` attribute to the `video` element.
```

jdwilkin4 commented on curriculum/challenges/english/blocks/workshop-html-video-player/68f1f8c248306da204655117.md:
```diff
@@ -0,0 +1,53 @@
+    loop
+    controls
+  >
+--fcc-editable-region--
```
```suggestion
--fcc-editable-region--
  <video width="640" loop controls>
--fcc-editable-region--
```

jdwilkin4 commented on curriculum/challenges/english/blocks/workshop-html-video-player/68f1fac259a4afabb478b960.md:
```diff
@@ -0,0 +1,59 @@
+
+# --hints--
+
+You have a `source` element.
```
```suggestion
You should have a `source` element inside of your `video` element.
```

jdwilkin4 commented on curriculum/challenges/english/blocks/workshop-html-video-player/68f268880219c2485bd71d2a.md:
```diff
@@ -0,0 +1,55 @@
+  <source
+  
+  />
+--fcc-editable-region--
```
```suggestion
--fcc-editable-region--
  <source />
--fcc-editable-region--
```

jdwilkin4 commented on curriculum/challenges/english/blocks/workshop-html-video-player/68f27139bddf9f6bf2821acf.md:
```diff
@@ -0,0 +1,124 @@
+
+Now, add the `source` element with the self-closing tag. Next, add the `src` attribute
+and the value `https://cdn.freecodecamp.org/curriculum/labs/mapmethod.mov`.
+Finally, add the `type` attribute and the value `video/quicktime`.
```
```suggestion
Below your third `source` element, add a fourth `source` element. Your fourth `source` element should have a `src` attribute with the value `https://cdn.freecodecamp.org/curriculum/labs/mapmethod.mov` and `type` attribute with the value `video/quicktime `.
```

jdwilkin4 reviewed (commented):
Went through the project locally and left some comments 👍🏾

mtrivera committed:
docs(curriculum): Apply suggestions from code review

Enhance grammar and vocabulary of descriptions. Improve learning experience for students.

Co-authored-by: Jessica Wilkins  <67210629+jdwilkin4@users.noreply.github.com>

ilenia-magoni commented on curriculum/challenges/english/blocks/workshop-html-video-player/68f1f57723debf9383c75947.md:
````diff
@@ -0,0 +1,51 @@
+
+```js
+const video = document.querySelector('video');
+assert.isNotNull(video?.getAttribute('loop'));
````
there is a comment to solve here that you missed

ilenia-magoni commented on curriculum/challenges/english/blocks/workshop-html-video-player/68f1f57723debf9383c75947.md:
````diff
@@ -0,0 +1,51 @@
+
+```js
+const video = document.querySelector('video');
+assert.isNotNull(video?.getAttribute('loop'));
````
```suggestion
assert.isTrue(video?.hasAttribute('loop'));
```

ilenia-magoni commented on curriculum/challenges/english/blocks/workshop-html-video-player/68f1f371fefd448b4f70b898.md:
````diff
@@ -0,0 +1,47 @@
+
+```js
+const video = document.querySelector('video');
+assert.isNotNull(video?.getAttribute('width'), '640');
````
```suggestion
assert.equal(video?.getAttribute('width'), '640');
```

ilenia-magoni commented on curriculum/challenges/english/blocks/workshop-html-video-player/68f1f57723debf9383c75947.md:
```diff
@@ -0,0 +1,51 @@
+
+The `loop` attribute will restart the video once
+playback is completed. Think of an internet meme that
+repeats playback. Omitting the `loop` atrtibute
```
```suggestion
repeats playback. Omitting the `loop` attribute
```

ilenia-magoni commented on curriculum/challenges/english/blocks/workshop-html-video-player/68f26a5f79e97c5032c10642.md:
````diff
@@ -0,0 +1,63 @@
+
+```js
+const source = document.querySelector('source');
+assert.strictEqual(source?. getAttribute('type'), 'video/mp4');
````
```suggestion
assert.strictEqual(source?.getAttribute('type'), 'video/mp4');
```

majestic-owl448 reviewed (commented):
the formatting of each attribute on it's own line does not look like something we use in other workshops, so maybe that would need to be adjusted.

but apart from that we may be almost there

while we also don't usually break sentences in multiple lines, that is not user facing so it's less important

mtrivera committed:
style(curriculum): Apply suggestions from code review

Fix spelling errors, advise assert styles, and attribute markup direction

Co-authored-by: Ilenia <26656284+ilenia-magoni@users.noreply.github.com>

mtrivera commented:
> the formatting of each attribute on it's own line does not look like something we use in other workshops, so maybe that would need to be adjusted.
> 
> but apart from that we may be almost there
> 
> while we also don't usually break sentences in multiple lines, that is not user facing so it's less important

I noticed in the `Hotel Feedback form` workshop it varies

```
id: 67a51d3a8a6fe123b77b0c6e
title: Step 15
challengeType: 0
dashedName: step-15
```

```html
<input type="text" id="full-name" name="name" placeholder="e.g., John Doe" required>
<input
placeholder="example@email.com"
            required
            id="email"
            type="email"
            name="email"
          />         
```

I noticed some inline and others on new lines. If  an `attribute=value` spans a long line, I'll put it on a new line. Would that work?

mtrivera commented:
> you can leave the formatting as is, confronting with other projects, it's fine
> 
> you will need to update with the latest version of `main` to resolve the failing test

@jdwilkin4 Mentioned the failing test to me. 
I changed the `blockType` to `blockLabel` w/o pulling the latest `main`.
Can this be resolved without using `push`? 
My fork doesn't have a `main` branch, but  `staging`.
Tried the following:
1. git co staging > git pull > git co html-video-workshop-841 > tests fail
2. git co html-video-workshop-841 > git pull > tests fail
3.  git pull --rebase upstream main > 
```
From https://github.com/freeCodeCamp/freeCodeCamp
 * branch                  main       -> FETCH_HEAD
   88fa4039b9..177dc4d668  main       -> upstream/main
Already up to date.
```
git co html-video-workshop-841 > pnpm clean-and-develop > tests fail

mtrivera committed:
fix(tools): Resolve blockLabel error

mtrivera committed:
test: Add h1 exists test step 1

mtrivera committed:
test: Change assert to exists check step 2

mtrivera committed:
test: Change assert to strictEqual check step 3

mtrivera committed:
test: Check source is child of video step 8

Dario-DC commented on client/i18n/locales/english/intro.json:
```diff
@@ -2105,6 +2105,13 @@
         ]
       },
+      "workshop-html-video-player": {
+        "title": "Build a HTML Video Player",
```
```suggestion
        "title": "Build an HTML Video Player",
```

Dario-DC commented on curriculum/challenges/english/blocks/workshop-html-video-player/68f18724d71db1dda8d6fec7.md:
```diff
@@ -0,0 +1,48 @@
+
+# --hints--
+
+You should have a `h1` element.
```
```suggestion
You should have an `h1` element.
```

Dario-DC commented on curriculum/challenges/english/blocks/workshop-html-video-player/68f1f993460f06a64c033061.md:
```diff
@@ -0,0 +1,53 @@
+the first video frame is shown during the download phase.
+
+Now, add the `poster` attribute and the 
+value `https://cdn.freecodecamp.org/curriculum/labs/past-event2.jpg`.
```
```suggestion
Now, add the `poster` attribute with the value `https://cdn.freecodecamp.org/curriculum/labs/past-event2.jpg` to your `video` element.
```

Dario-DC commented on curriculum/challenges/english/blocks/workshop-html-video-player/68f1f993460f06a64c033061.md:
```diff
@@ -0,0 +1,53 @@
+# --hints--
+
+Your `video` element should have the `poster` attribute and
+the value `https://cdn.freecodecamp.org/curriculum/labs/past-event2.jpg`.
```
```suggestion
Your `video` element should have the `poster` attribute with the value `https://cdn.freecodecamp.org/curriculum/labs/past-event2.jpg`.
```

Dario-DC commented on curriculum/challenges/english/blocks/workshop-html-video-player/68f26d52b13e0f5b6e2d2e09.md:
```diff
@@ -0,0 +1,72 @@
+VP8, VP9, and AV1, and audio codecs such as Vorbis and Opus, making it a popular
+choice for HTML5 video and audio elements.
+
+Below your first `source` element, add another `source` element. Your second `source` element should have a `src` attribute with the value `https://cdn.freecodecamp.org/curriculum/labs/mapmethod.webm` and `type` attribute with the value `video/webm`.
```
```suggestion
Below your first `source` element, add another `source` element and give it a `src` attribute with the value `https://cdn.freecodecamp.org/curriculum/labs/mapmethod.webm` and a `type` attribute with the value `video/webm`.
```

jdwilkin4 committed:
Apply suggestions from code review

Co-authored-by: Dario <105294544+Dario-DC@users.noreply.github.com>

mtrivera reviewed (commented):
LGTM

jdwilkin4 commented:
@mtrivera 

Have you pulled down the latest changes from main? Because I am running your PR locally and still see the old structure. 
Please make sure follow the suggestions here on updating your branch because your changes should target the new RWD v9 cert.

https://github.com/freeCodeCamp/freeCodeCamp/pull/62866#issuecomment-3485582389

mtrivera committed:
Merge branch 'main' of https://github.com/freeCodeCamp/freeCodeCamp into html-video-workshop-841

mtrivera committed:
chore(api): Support responsive-web-design-v9 cert

mtrivera committed:
Merge branch 'main' of https://github.com/freeCodeCamp/freeCodeCamp into html-video-workshop-841

mtrivera committed:
Merge branch 'main' of https://github.com/freeCodeCamp/freeCodeCamp into html-video-workshop-841

mtrivera committed:
style(api): Revert formatting changes to intro.json

majestic-owl448 commented on client/src/pages/learn/full-stack-developer/workshop-html-video-player/index.md:
the path of this file should be `client/src/pages/learn/responsive-web-design-v9/workshop-html-video-player/index.md`, please move it appropriately

majestic-owl448 commented on client/src/pages/learn/full-stack-developer/workshop-html-video-player/index.md:
```diff
@@ -0,0 +1,9 @@
+---
+title: Introduction to the Build an HTML Video Player
+block: workshop-html-video-player
+superBlock: full-stack-developer
```
```suggestion
superBlock: responsive-web-design-v9
```

majestic-owl448 commented on curriculum/challenges/english/blocks/workshop-html-video-player/68f268880219c2485bd71d2a.md:
```diff
@@ -0,0 +1,52 @@
+To specify the media resource for the video, you will need to add the `src` attribute to the `source` element.
+
+Add the `src` attribute with the value
+`https://cdn.freecodecamp.org/curriculum/labs/what-is-the-map-method-and-how-does-it-work.mp4`.
```
```suggestion
Add the `src` attribute with the value `https://cdn.freecodecamp.org/curriculum/labs/what-is-the-map-method-and-how-does-it-work.mp4`.
```

majestic-owl448 commented on curriculum/challenges/english/blocks/workshop-html-video-player/68f26a5f79e97c5032c10642.md:
```diff
@@ -0,0 +1,63 @@
+You will use the `type` attribute to specify the `video/mp4` MIME type.
+
+MIME (Multipurpose Internet Mail Extensions) is a standard to describe documents
+in other forms besides ASCII text, for example, audio, video, and images.
```
```suggestion
MIME (Multipurpose Internet Mail Extensions) is a standard to describe documents in other forms besides ASCII text, for example, audio, video, and images.
```

majestic-owl448 commented on curriculum/challenges/english/blocks/workshop-html-video-player/68f26d52b13e0f5b6e2d2e09.md:
```diff
@@ -0,0 +1,72 @@
+WebM is an open-source audiovisual media file format developed by Google,
+primarily designed for web-based media content. It supports video codecs like 
+VP8, VP9, and AV1, and audio codecs such as Vorbis and Opus, making it a popular
+choice for HTML5 video and audio elements.
```
```suggestion
WebM is an open-source audiovisual media file format developed by Google, primarily designed for web-based media content. It supports video codecs like VP8, VP9, and AV1, and audio codecs such as Vorbis and Opus, making it a popular choice for HTML5 video and audio elements.
```

majestic-owl448 committed:
Merge branch 'main' into html-video-workshop-841

majestic-owl448 committed:
remove formatting changes

majestic-owl448 committed:
index.md

majestic-owl448 committed:
remove curriculum/structure/superblocks/full-stack-developer.json

majestic-owl448 reviewed (commented):
resolved conflicts and migrated PR to rwd v9

mtrivera committed:
Apply suggestions from code review

Change description lists to paragraphs.

Co-authored-by: majestic-owl448 <26656284+majestic-owl448@users.noreply.github.com>

mtrivera committed:
Apply suggestions from code review

Grammar improvements and add exist element tests. Remove `/` from void elements for consistency.

Co-authored-by: majestic-owl448 <26656284+majestic-owl448@users.noreply.github.com>

Dario-DC commented on curriculum/challenges/english/blocks/workshop-html-video-player/68f1f57723debf9383c75947.md:
```diff
@@ -0,0 +1,45 @@
+
+The `loop` attribute will restart the video once playback is completed. Think of an internet meme that repeats playback. Omitting the `loop` attribute will make the video playback once.
+
+The `loop` attribute is a boolean attribute, it does not need a value.
```
```suggestion
The `loop` attribute is a boolean attribute and does not need a value.
```

mtrivera committed:
Apply suggestions from code review

Grammar improvements and add to responsive-web-design-v9.json

Co-authored-by: Dario <105294544+Dario-DC@users.noreply.github.com>

Dario-DC reviewed (approved):
LGTM 🎉
